pkg/api/defaults-kubelet_test.go

```diff
@@ -499,6 +519,15 @@ func TestKubeletIPMasqAgentEnabledOrDisabled(t *testing.T) {
 		t.Fatalf("got unexpected '--non-masquerade-cidr' kubelet config value %s, the expected value is %s",
 			k["--non-masquerade-cidr"], DefaultNonMasqueradeCIDR)
 	}
+
+	// No ip-masq-agent addon configuration specified, --non-masquerade-cidr should be 0.0.0.0/0
```

Comment:
@aramase @CecileRobertMichon Actually, this additional UT is not what we want. If I add this coverage to the commit prior to #2393 (`$ git checkout 6e76fd9e69a7ee8aa61d2d3c559919a460bad5ed`) this test fails:

```
got unexpected '--non-masquerade-cidr' kubelet config value 10.240.0.0/12, the expected value is 0.0.0.0/0
```

We want to identify exactly what cluster configuration + defaults enforcement *was* working prior to #2393, and prove that this fix restores that functionality.

Reply:
@jackfrancis That is because in the unit tests `ip-masq-agent` was never enabled. So these tests will fail in all versions prior to this. But going forward it will help us spot such issues.

Reply:
If this is truly a fix, then something must have broken in #2393 that can be expressed as a UT, right?

Reply:
This was never unit tested before. There are individual unit tests for `setKubeletConfig` and `setAddonConfig`, but none which check the desired result after running both.

Reply:
With this change the ip-masq-agent enabled/disabled isn't really dependent on `setAddonConfig` anymore as the check to see if ip-masq is enabled/disabled is done outside of that while setting non-masq-cidr. So I feel this unit test is appropriate. This unit test addition is essentially increasing the coverage for the path we didn't check before.